### PR TITLE
use the new streamer

### PR DESCRIPTION
**Pull request by jorendumoulin**
Awaiting the merge of the streamer refactoring, this PR updates the test to be compatible with additional assertions regarding the addresses which need to be aligned with a certain bank.

It also fixes some smaller stuff with the tests + adds a lot more complicated tests to the `streamer-tcdm` sytem, now including bank conflicts and stalls from the accelerator!

### CODE REVIEW

File: tests/cocotb/snax_util.py [hunk 1]
```diff
@@ -231,7 +235,7 @@ async def reg_clr(dut) -> None:
     dut.io_csr_req_bits_addr_i.value = 0
     dut.io_csr_req_bits_write_i.value = 0
     dut.io_csr_req_valid_i.value = 0
-    await clock_and_wait(dut)
+    # await clock_and_wait(dut)
```
Comment (rgantonio) [inline tests/cocotb/snax_util.py:238]: rm -f haha

File: tests/cocotb/test_stream_tcdm.py [hunk 2]
```diff
@@ -213,7 +213,7 @@ async def stream_tcdm_dut(dut):
 
     # Wait for the rising edge of the valid
     # From there we can continuously stream for ever clock cycle
-    await with_timeout(RisingEdge(dut.stream2acc_data_0_valid_o), 100, "ns")
+    # await with_timeout(RisingEdge(dut.stream2acc_data_0_valid_o), 100, "ns")
```
Comment (rgantonio) [inline tests/cocotb/test_stream_tcdm.py:216]: rm -f

File: tests/cocotb/test_stream_tcdm.py [hunk 3]
```diff
@@ -267,6 +274,203 @@ async def stream_tcdm_dut(dut):
...
+    for _ in range(10):
+        await snax_util.clock_and_wait(dut)
+    dut.rst_ni.value = 0
+    await snax_util.clock_and_wait(dut)
+    await snax_util.clock_and_wait(dut)
+    dut.rst_ni.value = 1
+    for _ in range(10):
+        await snax_util.clock_and_wait(dut)
+
+    cocotb.log.info("Writer-reader test - with contention")
```
Comment (rgantonio) [inline tests/cocotb/test_stream_tcdm.py:298]: Niceee!

**Review by rgantonio — APPROVED** [review]
I am amazed that you understood the tests I made so fast! You're so good @jorendumoulin !!!

BTW I presume that you will update the `Bender.yml` file later after the changes have been pushed to the streamer yes?

**Comment by jorendumoulin** [comment]
> BTW I presume that you will update the `Bender.yml` file later after the changes have been pushed to the streamer yes?

Yes exactly! The snax-dev setup is so nice to work with, great work on this 😄

**Comment by rgantonio** [comment]
@jorendumoulin I presume the streamers are okay now? 😄 so maybe we can update this? I like the fixes haha

**Comment by jorendumoulin** [comment]
Yes! Once the updates in `snax-streamer` get merged (kuleuven-micas/snax-streamer#29). The bender here can be changed to the `main` branch and it is ready to be merged!

**Comment by rgantonio** [comment]
@jorendumoulin I or you squash and merge? 😄

**Pull request merged**